tracer/src/Datadog.Trace/Agent/DiscoveryService/DiscoveryService.cs

```diff
@@ -0,0 +1,93 @@
+// <copyright file="DiscoveryService.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Logging;
+using Datadog.Trace.Vendors.Newtonsoft.Json;
+using Datadog.Trace.Vendors.Newtonsoft.Json.Linq;
+
+namespace Datadog.Trace.Agent.DiscoveryService;
+
+internal class DiscoveryService : IDiscoveryService
```

Comment:
This looks like it would be useful outside of the Debugger context 🙂

Reply:
This is essentially what we need to detect if we can enable the "stats in tracer" feature! 😄

Reply:
Yeps, I believe this is also what's needed for the AppSec to determine whether they can enable the RCM-dependent features. /cc @robertpi

Reply:
Also useful for CI Visibility to discover if the agent has support for the `evpproxy` (Event-Platform Proxy) endpoint.

Reply:
Will also be necessary for the Data Streams Monitoring work I'm looking at shortly

Reply:
I like this DiscoveryService class, but it will need to be more generic to support multiple products/features and endpoints.

Reply:
Totally agreed, we had already discussed some ways to make it more generic / prettier when the time comes to use it across products. Seems like we all agree :) Since this PR already has a **lot** of moving parts, do we all agree to do that refactoring as a separate PR?